net/cgi-io/src/main.c

```diff
@@ -782,9 +786,7 @@ main_backup(int argc, char **argv)
 
 		fflush(stdout);
 
-		do {
-			len = splice(fds[0], NULL, 1, NULL, 4096, SPLICE_F_MORE);
-		} while (len > 0);
+		while (splice(fds[0], NULL, 1, NULL, READ_BLOCK, SPLICE_F_MORE) == READ_BLOCK);
```

Comment:
Afaik a read size less than the requested size is no proper abort condition. Only a zero or negative return should terminate the loop.

Reply:
The logic is that if the size is less than the block then we are at the end of the read... But could happen that for some reason the other pipe is still sending data... 
So yes will drop this change